python/tests/test_metadata.py

```diff
@@ -572,6 +572,10 @@ def test_dont_skip_validation_other_codecs(self):
             ms.validate_and_encode_row({"int": 1})
             assert mocked_validate.call_count == 1
 
+    def test_zero_length(self):
+        ms = tskit.MetadataSchema({"codec": "json"})
+        assert ms.decode_row(b"") == {}
+
 
```

Comment:
We're not currently testing the interaction between the empty metadata fields and the defaults. Can you add a few tests with examples to make this explicit please? (This could easily be implemented in a way that bypasses adding the default values into the result)